src/Mod/Part/App/PartFeature.cpp

```diff
@@ -295,26 +289,14 @@ static TopoShape _getTopoShape(const App::DocumentObject *obj, const char *subna
         }
     }
 
-    if(_ShapeCache.getShape(obj,shape,subname)) {

```

Comment:
Be careful! This changes the semantic because it makes a difference calling `_ShapeCache.getShape(obj,shape,subname)` or not.

Reply:
reverted.